src/affine/__init__.py

```diff
@@ -328,18 +302,22 @@ def __array__(self, dtype=None, copy=None):
     def __str__(self) -> str:
         """Concise string representation."""
         return (
-            "|{: .2f},{: .2f},{: .2f}|\n"
-            "|{: .2f},{: .2f},{: .2f}|\n"
-            "|{: .2f},{: .2f},{: .2f}|"
-        ).format(*self)
+            f"|{self.a: .2f},{self.b: .2f},{self.c: .2f}|\n"
+            f"|{self.d: .2f},{self.e: .2f},{self.f: .2f}|\n"
+            f"|{self.g: .2f},{self.h: .2f},{self.i: .2f}|"
+        )
 
     def __repr__(self) -> str:
         """Precise string representation."""
-        # fmt: off
-        return ("Affine({!r}, {!r}, {!r},\n"
-                "       {!r}, {!r}, {!r})"
-        ).format(*self[:6])
-        # fmt: on
+        return (
+            f"Affine({self.a!r}, {self.b!r}, {self.c!r},\n"
+            f"       {self.d!r}, {self.e!r}, {self.f!r})"
+        )
+
+    @cached_property
+    def precision(self):
+        """Numerical precision of comparison methods."""
+        return EPSILON
```

Comment:
Making precision a property, not an attribute, makes all the uses of astuple() much cleaner.